nixos/modules/services/networking/powerdns.nix

```diff
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.powerdns;
+  configDir = pkgs.writeTextDir "pdns.conf" "${cfg.extraConfig}";
+in {
+  options = {
+    services.powerdns = {
+      enable = mkOption {
+        default = false;
+        description = "Whether to enable Powerdns domain name server.";
+      };
+
+      package = mkOption {
+        default = pkgs.powerdns;
+        defaultText = "pkgs.powerdns";
+        description = "Which powerdns derivation to use";
+        type = types.package;
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "launch=bind";
+        description = ''
+          Extra lines to be added verbatim to pdns.conf.
+          Powerdns will chroot to /var/lib/powerdns.
+          So any file, powerdns is supposed to be read,
+          should be in /var/lib/powerdns and needs to specified
+          relative to the chroot.
+        '';
+      };
+    };
+  };
+
+  config = mkIf config.services.powerdns.enable {
+    systemd.services.powerdns = {
+      unitConfig.Documentation = "man:pdns_server(1) man:pdns_control(1)";
+      description = "Powerdns name server";
+      wantedBy = [ "multi-user.target" ];
+      after = ["network.target" "mysql.service" "postgresql.service" "openldap.service"];
+
+      preStart = ''
+        mkdir -p /var/lib/powerdns/
+        # powerdns creates control socket after chroot
+        # this way pdns_control can find the control socket
+        ln -sTf /var/lib/powerdns/pdns.controlsocket /var/run/pdns.controlsocket
```

Comment:
Imho it would be better to pass `--with-socketdir` to configure so we don't need to symlink (the symlink will not be cleaned up automatically..)